subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -330,7 +317,7 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv)
 	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_ACTIVE)) {
 		atomic_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULE_PENDING);
 		return false;
-	} else if (k_work_is_pending(&ext_adv->work)) {
+	} else if (!force && k_work_is_pending(&ext_adv->work)) {
```

Comment:
Could you try this instead of the `force` flag:
```suggestion
	} else if (k_work_busy_get(&ext_adv->work) & (K_WORK_QUEUED | K_WORK_DELAYED)) {
```

Reply:
There is a situation where the application layer sends mesh messages multiple times. If the first work is running but not yet set ADV_FLAG_ACTIVE, the second call will trigger k_work_submit.

Reply:
for example LOG_DBG with config:LOG_MODE_IMMEDIATE.

Reply:
> If the first work is running but not yet set ADV_FLAG_ACTIVE, the second call will trigger k_work_submit.

Shouldn't `(K_WORK_QUEUED | K_WORK_DELAYED)` flags protect exactly against this?

Reply:
i'm first work is running, not queue, is `K_WORK_RUNNING`

Reply:
You can imagine that when the work is running, but the flag of `ADV_FLAG_ACTIVE ` has not been set, and at same time the application layer task can still call `schedule_send` and finally `k_work_submit` again.

Reply:
Another compromise